File: lld_game_design_primer/minesweeper/Approach_1/Cell.py
class Cell:
    def __init__(self, r, c):
        self._isBomb = False
        self._number = 0
        self._isExposed = False
        self._isFlagged = False

        self._row = r
        self._column = c

    def setRowAndColumn(self, r, c):
        self._row = r
        self._column = c

    def setBomb(self):
        self._isBomb = True
        self._number = -1

    def flag(self):
        self._isFlagged = True

    def unflag(self):
        self._isFlagged = False

    def isBlank(self):
        return self._number == 0

    def isNumber(self):
        return self._number > 0

    def isExposed(self):
        return self._isExposed

    def expose(self):
        self._isExposed = True

    def getNumber(self):
        return self._number

    def setNumber(self, number):
        self._number = number

    def getRow(self):
        return self._row

    def getColumn(self):
        return self._column

    def isBomb(self):
        return self._isBomb

    

File: lld_game_design_primer/minesweeper/Approach_1/Cell.java
package lld_game_design_primer.minesweeper.Approach_1;

public class Cell {
    private int row;
    private int column;
    private boolean isBomb;
    private int number;
    private boolean isExposed;
    private boolean isFlagged;

    public Cell(int r, int c) {
        row = r;
        column = c;
    }

    public void setRowAndColumn(int r, int c) {
        row = r;
        column = c;
    }

    public void setBomb() {
        isBomb = true;
        number = -1;
    }

    public void flag() {
        isFlagged = true;
    }

    public void unflag() {
        isFlagged = false;
    }

    public boolean isBlank() {
        return number == 0;
    }

    public boolean isNumber() {
        return number > 0;
    }

    public boolean isExposed() {
        return isExposed;
    }

    public void expose() {
        isExposed = true;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public int getRow() {
        return row;
    }

    public int getColumn() {
        return column;
    }

    public boolean isBomb() {
        return isBomb;
    }
}

